multiapps-controller-web/src/main/java/org/cloudfoundry/multiapps/controller/web/configuration/FlowableConfiguration.java

```diff
@@ -34,6 +36,8 @@
 
     private static final int ASYNC_JOB_ACQUIRE_WAIT_TIME_IN_MILLIS = (int) TimeUnit.SECONDS.toMillis(3);
     private static final int JOB_EXECUTOR_LOCK_TIME_IN_MILLIS = (int) TimeUnit.MINUTES.toMillis(30);
+    private static final long JOB_CANCEL_WAIT_TIME_IN_MILLIS = TimeUnit.MINUTES.toMillis(1);
+    private static final long JOB_TIMEOUT_TIME_IN_MILLIS = JOB_EXECUTOR_LOCK_TIME_IN_MILLIS - 2 * JOB_CANCEL_WAIT_TIME_IN_MILLIS;
```

Comment:
Why is this formula used?

Reply:
So we want to check if a job will expire a little before it actually expires. So 30 mins minus 1 minute, in which the thread could be sleeping and 1 minute more for prevention of exactly canceling it at 30 mins